src/processors/operation-processor.ts

```diff
@@ -11,9 +29,61 @@ export default class OperationProcessor<ResourceT = Resource> {
     return this.resourceClass && op.ref.type === this.resourceClass.name;
   }
 
-  execute(op: Operation): Promise<ResourceT | ResourceT[] | void> {
+  async execute(op: Operation): Promise<ResourceT | ResourceT[] | void> {
     const action: string = op.op;
-    return this[action] && this[action].call(this, op);
+    let result = this[action] && await this[action].call(this, op);
+
+    return this.convertToResources(op.ref.type, result);
+  }
+
+  getAttributes(attributes, fields, type): string[] {
+    if (Object.entries(fields).length === 0 && fields.constructor === Object) {
+      return attributes;
+    }
+
+    return attributes.filter(attribute =>
+      fields[pluralize(type)].includes(attribute)
+    );
+  }
+
+  get computedPropertyNames() : string[] {
+    return Object.keys(this.attributes);
+  }
+
+  async getComputedProperties(record: HasId) {
+    return promiseHashMap(this.attributes, (key) => this.attributes[key].call(this, record));
+  }
+  async getRelationships(record: HasId) {
+    return promiseHashMap(this.relationships, (key) => this.relationships[key].call(this, record));
+  }
+
+  async convertToResources(type: string, records: HasId[] | HasId) {
+    if (Array.isArray(records)) {
+      return Promise.all(records.map(record => {
+        return this.convertToResources(type, record);
+      }));
+    }
+
+    const record = {...records};
+    const id = record.id;
+    const [computedAttributes, relationships] = await Promise.all([
```

Comment:
Here's where we calculate the computed properties and relationships for each resource.